crates/resolver/src/lookup.rs

```diff
@@ -35,11 +34,20 @@ use crate::proto::dnssec::Proven;
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct Lookup {
     query: Query,
-    records: Arc<[Record]>,
+    message: Arc<Message>,
     valid_until: Instant,
 }
 
 impl Lookup {
+    /// Create a new Lookup from a complete DNS Message.
+    pub fn new(query: Query, message: Message, valid_until: Instant) -> Self {
+        Self {
+            query,
```

Comment:
Does it still make sense to store the query here if it's also stored in the `message`?

Reply:
Let's leave it in for now. It would be useful if you added some of the caveats from Discord in comments here to make this a little more obvious.